test/test.test.js

```diff
@@ -2155,3 +2156,149 @@ test('format user', function (t) {
     t.equal(user.baz, 'bar')
   })
 })
+
+test('expose decode token for plugin extension', function (t) {
+  t.plan(3)
+
+  const fastify = Fastify()
+  fastify.register(jwt, { secret: 'test' })
+
+  fastify.post('/sign', async function (request, reply) {
+    const token = await reply.jwtSign(request.body)
+    return { token }
+  })
+
+  fastify.get('/check-decoded-token', async function (request, reply) {
+    const decodedToken = await request.jwtDecode()
+    return reply.send(decodedToken)
+  })
+
+  fastify.get('/check-decoded-token-callback', function (request, reply) {
+    request.jwtDecode((err, decodedToken) => {
+      if (err) {
+        return reply.send(err)
+      }
+      return reply.send(decodedToken)
+    })
+  })
+
+  t.test('should decode token without verifying', async function (t) {
+    t.plan(2)
+
+    const signResponse = await fastify.inject({
+      method: 'post',
+      url: '/sign',
+      payload: { foo: 'bar' }
+    })
+    const token = JSON.parse(signResponse.payload).token
+    t.ok(token)
+
+    const decodeResponse = await fastify.inject({
+      method: 'get',
+      url: '/check-decoded-token',
+      headers: {
+        authorization: `Bearer ${token}`
+      }
+    })
+    const decodedToken = JSON.parse(decodeResponse.payload)
+    t.equal(decodedToken.foo, 'bar')
+  })
+
+  t.test('should decode token with callback', async function (t) {
+    t.plan(2)
+
+    const signResponse = await fastify.inject({
+      method: 'post',
+      url: '/sign',
+      payload: { foo: 'bar' }
+    })
+    const token = JSON.parse(signResponse.payload).token
+    t.ok(token)
+
+    const decodeResponse = await fastify.inject({
+      method: 'get',
+      url: '/check-decoded-token-callback',
+      headers: {
+        authorization: `Bearer ${token}`
+      }
+    })
+    const decodedToken = JSON.parse(decodeResponse.payload)
+    t.equal(decodedToken.foo, 'bar')
+  })
+
+  t.test('should handle decode error', async function (t) {
+    t.plan(1)
+
+    const decodeResponse = await fastify.inject({
+      method: 'get',
+      url: '/check-decoded-token',
+      headers: {}
+    })
+    const decodedToken = JSON.parse(decodeResponse.payload)
+    t.equal(decodedToken.statusCode, 401)
+  })
+})
+
+test('support extended config contract', function (t) {
+  t.plan(1)
+  const extConfig = {
+    sign: {
+      issuer: 'api.example.tld'
+    },
+    verify: {
+      issuer: 'api.example.tld'
```

Comment:
Checking the code, the `verify` option uses the `extractToken` parameter only.
The `issuer` is ignored.

I would add the `decode` parameter too